28-door-state-monitor/main.py

```diff
@@ -0,0 +1,52 @@
+from machine import I2C, Pin, Timer
+import notecard 
+from notecard import card, hub, note
+import time
+
+# Config
+# productUID = "com.my-company.my-name:my-project"
+
+# Global Variables for Interupt
+debounce_time=0
+sendDoorChange = False
+
+def doorChange():
+    print("Door", ("closed","open")[door.value()])
+    rsp = note.add(nCard,file = "door.qo", body = {"open": bool(door.value()), "closed": not door.value()}, port = 11, sync = True)  # Door is low when closed (pull up input)
+    print(rsp)
+    
+def doorCallback(pin):
+    global sendDoorChange, debounce_time
+    if (time.ticks_ms()-debounce_time) > 500:
+        sendDoorChange=True
+        debounce_time=time.ticks_ms()
+
+# Create I2C instance 
+i2c=I2C(0, freq=400000)
+
+# Connect to Notecard   
+nCard = notecard.OpenI2C(i2c, 0, 0) 
+
+# Set ProductUID 
+
+hub.set(nCard, product=productUID, mode="minimum") # We use minimum so we can sync and reset PIR count together
+
+# Setup templates
+
+rsp = note.template(nCard, file="door.qo", body={"open": True, "closed": True}, port = 11, compact = True)
+print(rsp)
+
+door = Pin(26, Pin.IN, Pin.PULL_UP)
+
+door.irq(trigger=Pin.IRQ_FALLING|Pin.IRQ_RISING, handler=doorCallback)
+
+
+while True:
+    machine.disable_irq
```

Comment:
Was this imported correctly? I’m getting warnings that `machine` is not defined so maybe just quickly check this.

Reply:
If you are looking in VSCode it is because it doesn't know about MicroPython, you will see in line 1 it also complains about machine